inst/include/RcppSimdJson/deserialize/simplify.hpp

```diff
@@ -75,11 +78,11 @@ inline SEXP simplify_data_frame(simdjson::dom::array array,
 
 
 template <Type_Policy type_policy, utils::Int64_R_Type int64_opt, Simplify_To simplify_to>
-inline SEXP dispatch_simplify_array(simdjson::dom::array array,
+inline SEXP dispatch_simplify_array(simdjson::ondemand::array array,
                                     SEXP                 empty_array,
                                     SEXP                 empty_object,
                                     SEXP                 single_null) {
-    if (std::size(array) == 0) {
+    if (array.count_elements() == 0) {
```

Comment:
This is inefficient with On Demand. Consider the scenario where the array has many elements. Then counting them 

Consider using something like `if(!array.rewind()) {...}`.

Reply:
Ok, I will change it everywhere.